spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,132 @@
+  before(:all) do
+    FactoryBot.create_list(:restaurant, 10)
+  end
+
+  describe 'GET /restaurants' do
+    before do
+      get restaurants_path
+    end
+
+    it 'return https success' do
+      expect(response).to have_http_status(200)
+    end
+
+    it 'has the right objects count' do
+      arr = response_body
+      expect(arr.count).to eq(Restaurant.all.count)
+    end
+
+    it 'contains mandatory attributes' do
+      arr = response_body
+      arr.each do |json_rest|
+        expect(json_rest.keys).to include('id', 'name', 'cuisine', 'accept10bis')
+      end
+    end
+  end
+
+  describe 'GET /restaurant/:id' do
+    subject { Restaurant.all.sample }
+    it 'return https success' do
+      get restaurant_path(subject.id)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns :not_found status on invalid uuid' do
+      get restaurant_path('this-is-not-a-uuid')
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it 'returns the correct record' do
+      get restaurant_path(subject.id)
+      restaurant_matcher(subject, response_body)
+    end
+  end
+
+  describe 'POST /restaurants' do
+    it 'returns http success' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
+      expect(response).to have_http_status(:created)
+    end
+
+    it 'creates new record in the database' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
+      uuid = response_body['id']
+      rest = Restaurant.find_by(id: uuid)
+      expect(rest).to_not be_nil
+      restaurant_matcher(rest, response_body)
+    end
+
+    it 'returns error when params are invalid' do
+      invalid_attr = FactoryBot.attributes_for(:restaurant, :invalid_record)
+      post '/restaurants', params: invalid_attr
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+  end
+
+  describe 'PUT /restaurant/:id' do
+    subject { Restaurant.all.sample }
+
+    it 'returns https success' do
+      put restaurant_path(subject.id), params: FactoryBot.attributes_for(:restaurant)
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'returns error when params are invalid' do
+      invalid_attr = FactoryBot.attributes_for(:restaurant, :invalid_record)
+      put restaurant_path(subject.id), params: invalid_attr
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+
+    it 'updates the record fields' do
+      new_rest = FactoryBot.build(:restaurant, id: subject.id)
+      put restaurant_path(subject.id), params: new_rest.attributes
+      rest = Restaurant.find(subject.id)
+      restaurant_matcher(rest, new_rest)
+    end
+  end
+
+  describe 'DELETE /restaurant/:id' do
+    subject { Restaurant.all.sample }
+
+    it 'returns http success' do
+      delete restaurant_path(subject.id)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'fails to find the record in DB' do
```

Comment:
this test relay on the previous test - let's not do it.
test never should be relay on another test, each test need to be isolated